.github/workflows/pull_request_full_build.yml

```diff
@@ -67,16 +62,29 @@ jobs:
         run: |
           wget https://raw.githubusercontent.com/ballerina-platform/ballerina-release/master/dependabot/resources/extensions.json
           wget https://raw.githubusercontent.com/ballerina-platform/ballerina-distribution/2201.0.x/gradle.properties
+
       - name: Clone Modules
         run: |
           for module_name in $(jq -r '.standard_library| .[] | select(.level==${{ matrix.level }}) | .name' extensions.json); \
           do git clone https://github.com/ballerina-platform/${module_name}.git; \
           done
+
+      - name: Checkout non-default branch
+        run: |
+          for module_name in $(jq -r '.standard_library| .[] | select(.level==${{ matrix.level }}) | .name' extensions.json); \
```

Comment:
@niveathika shall we change this as below? As I know, we can't use for loops line by line in github actions shell.
```
for module_name in $(jq -r '.standard_library| .[] | select(.level==${{ matrix.level }}) | .name' extensions.json); do \
cd $module_name && \
git fetch origin && \
git checkout -t origin/update4 ||: && \
cd ..; \
done
```